pkg/aspect/clean/clean.go

```diff
@@ -165,3 +189,461 @@ func (c *Clean) Run(_ *cobra.Command, _ []string) error {
+			// Keeping the bazel workspace for the current repo will mean faster build times for that repo.
+			// Dividing by 2 so that the current workspace will be listed later to the user.
+			comparator = comparator / 2
+		}
+
+		bazelDir.comparator = comparator
+
+		out <- bazelDir
+	}
+
+	waitGroup.Done()
+}
+
+func (c *Clean) deleteProcessor(
+	deleteQueue chan bazelDirInfo,
+	waitGroup *sync.WaitGroup,
+	sizeQueue chan<- float64,
+	errors chan<- error,
+) {
+	for bazelDir := range deleteQueue {
+
+		// We know that there will be an "external" folder that could be deleted in parallel.
+		// So we can move those folders to a tmp filepath and add them as seperate deletes that will
+		// therefore happen in parallel.
+		externalFolders, _ := ioutil.ReadDir(filepath.Join(bazelDir.path, "external"))
+		for _, folder := range externalFolders {
+			newPath := c.MoveFolderToTmp(bazelDir.path, folder.Name())
+
+			if newPath != "" {
+				waitGroup.Add(1)
+				deleteQueue <- bazelDirInfo{
+					path: newPath,
+					size: 0, // dont want to calculate the size for the subfolders. Just include the parent size
+				}
+			}
+		}
+
+		// otherwise certain files will throw a permission error when we try to remove them
+		if _, err := c.ChangeFolderPermissions(bazelDir.path); err != nil {
+			waitGroup.Done()
+			errors <- fmt.Errorf("failed to delete %q: %w", bazelDir.path, err)
+			continue
+		}
+
+		// remove entire folder
+		if err := os.RemoveAll(bazelDir.path); err != nil {
+			waitGroup.Done()
+			errors <- fmt.Errorf("failed to delete %q: %w", bazelDir.path, err)
+			continue
+		}
+
+		sizeQueue <- bazelDir.size
+
+		waitGroup.Done()
+	}
+}
+
+func (c *Clean) errorProcessor(errorQueue <-chan error, waitGroup *sync.WaitGroup, errors *errorSet) {
+	for err := range errorQueue {
+		errors.insert(err)
+	}
+	waitGroup.Done()
+}
+
+func (c *Clean) sizePrinter(sizeQueue <-chan float64, waitGroup *sync.WaitGroup) {
+	var totalSize float64 = 0
+
+	for size := range sizeQueue {
+		totalSize = totalSize + size
+	}
+
+	_, hRSpaceReclaimed, unit := c.makeBytesHumanReadable(totalSize)
+	fmt.Printf("Space reclaimed: %.2f%s\n", hRSpaceReclaimed, unit)
+
+	waitGroup.Done()
+}
+
+func (c *Clean) findBazelBaseDir() (string, string, error) {
+	cwd, err := os.Getwd()
+	if err != nil {
+		return "", "", fmt.Errorf("failed to find Bazel base dir: %w", err)
+	}
+
+	files, err := ioutil.ReadDir(cwd)
+	if err != nil {
+		return "", "", fmt.Errorf("failed to find Bazel base dir: %w", err)
+	}
+
+	for _, file := range files {
+
+		// bazel-bin, bazel-out, etc... will be symlinks, so we can eliminate non-symlinks immediately
+		if file.Mode()&os.ModeSymlink != 0 {
+			actualPath, err := os.Readlink(filepath.Join(cwd, file.Name()))
+			if err != nil {
+				return "", "", fmt.Errorf("failed to find Bazel base dir: %w", err)
+			}
+
+			if strings.Contains(actualPath, "bazel") && strings.Contains(actualPath, "/execroot/") {
```

Comment:
If we end up supporting Windows, not sure this is a good thing.
You may want to use https://pkg.go.dev/path#Split and then check if one of the parts of the path matches `execroot`.

Reply:
`path.Split` does not do this split. It simply splits the the directory from the filename. I have instead changed my approach to use `os.PathSeparator`. Hopefully that addresses your concerns?